File: Java Spring Boot Project/CompanyDirectory/src/main/java/com/mindthecode/CompanyDirectory/models/entities/Department.java
package com.mindthecode.CompanyDirectory.models.entities;

import javax.persistence.*;

@Entity
@Table(name = "Departments")
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    private String name;

    @ManyToOne
    private BusinessUnit businessUnit;


    public Department() {
    }

    public Department(String name, BusinessUnit businessUnit) {
        this.name = name;
        this.businessUnit = businessUnit;
    }

    public Department(long id, String name, BusinessUnit businessUnit) {
        this.id = id;
        this.name = name;
        this.businessUnit = businessUnit;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public BusinessUnit getBusinessUnit() {
        return businessUnit;
    }

    public void setBusinessUnit(BusinessUnit businessUnit) {
        this.businessUnit = businessUnit;
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/entities/department.dart
import 'package:company_directory_app/models/entities/business_unit.dart';

class Department {
  final int id;
  final String name;
  final BusinessUnit businessUnit;

  Department({this.id, this.name, this.businessUnit});

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(id: json['id'], name: json['name'], businessUnit: BusinessUnit.fromJson(json['businessUnit']));
  }
}
